python_modules/libraries/dagster-k8s/dagster_k8s/external_resource.py

```diff
@@ -0,0 +1,133 @@
+
+import kubernetes
+from dagster import OpExecutionContext
+from dagster._core.external_execution.context import (
+    ExternalExecutionOrchestrationContext,
+)
+from dagster._core.external_execution.resource import (
+    ExternalExecutionResource,
+)
+from dagster_externals import (
+    DAGSTER_EXTERNALS_ENV_KEYS,
+    ExternalExecutionExtras,
+)
+
+from .client import DagsterKubernetesClient, WaitForPodState
+
+
+def get_pod_name(run_id: str, op_name: str):
+    clean_op_name = op_name.replace("_", "-")
+    suffix = "".join(random.choice(string.digits) for i in range(10))
+    return f"dagster-{run_id[:18]}-{clean_op_name[:20]}-{suffix}"
+
+
+DEFAULT_TIMEOUT = 30
+
+
+class K8sPodExecutionResource(ExternalExecutionResource):
+    # config?
+
+    def run(
+        self,
+        context: OpExecutionContext,
+        image: str,
+        command: Union[str, Sequence[str]],
+        namespace: Optional[str],
+        env: Mapping[str, str],
+        extras: Optional[ExternalExecutionExtras] = None,
+    ) -> None:
+        client = DagsterKubernetesClient.production_client()
+
+        external_context = ExternalExecutionOrchestrationContext(context=context, extras=extras)
+
+        namespace = namespace or "default"
+        pod_name = get_pod_name(context.run_id, context.op.name)
+        context_config_map_name = pod_name
+
+        io_env = {
+            DAGSTER_EXTERNALS_ENV_KEYS["context_source"]: json.dumps(
+                {"path": "/mnt/dagster/context.json"}
+            ),
+            "STDOUT_MSG_SINK": "1",
+        }
+
+        context_config_map_body = kubernetes.client.V1ConfigMap(
+            metadata=kubernetes.client.V1ObjectMeta(
+                name=context_config_map_name,
+            ),
+            data={
+                "context.json": json.dumps(external_context.get_data()),
+            },
+        )
+        client.core_api.create_namespaced_config_map(namespace, context_config_map_body)
+
+        pod_body = kubernetes.client.V1Pod(
+            metadata=kubernetes.client.V1ObjectMeta(
+                name=pod_name,
+            ),
+            spec=kubernetes.client.V1PodSpec(
+                restart_policy="Never",
+                volumes=[
+                    {
+                        "name": "dagster-externals-context",
+                        "configMap": {
+                            "name": context_config_map_name,
+                        },
+                    }
+                ],
+                containers=[
+                    kubernetes.client.V1Container(
+                        name="execution",
+                        image=image,
+                        command=command,
+                        env=[
+                            {"name": k, "value": v}
+                            for k, v in {
+                                **self.get_base_env(),
+                                **io_env,
+                                **env,
+                            }.items()
```

Comment:
So I'm new to these massive slugs of k8s config code, but it strikes me as eminently unpluggable. Brief googling yields a bazillion options for configuration, and we would need to alter this code to support them.  Is there a way to have a pluggability point so a user could manage `V1Pod` creation in a pinch?

Reply:
thinking on this, will try something in for the MVP that we can change if we dont like